lib/src/controllers/calendar_controller.dart

```diff
@@ -0,0 +1,136 @@
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import 'dart:collection';
+
+import 'package:flutter_app/src/model/ntut/ntut_calendar_json.dart';
+import 'package:flutter_app/src/task/ntut/ntut_calendar_task.dart';
+import 'package:flutter_app/src/task/task_flow.dart';
+import 'package:flutter_app/src/util/language_util.dart';
+import 'package:get/get_core/get_core.dart';
+import 'package:get/get_instance/get_instance.dart';
+import 'package:get/get_rx/get_rx.dart';
+import 'package:get/get_state_manager/get_state_manager.dart';
+import 'package:table_calendar/table_calendar.dart';
+
+class CalendarController extends GetxController {
+  static CalendarController get to => Get.find();
+
+  final Map<DateTime, List<NTUTCalendarJson>> knownHolidays = {
+    // TODO: Define the source of holidays in a correct way.
+    // FIXME: The type of holiday should not be `NTUTCalendarJson`.
+  };
+
+  final knownSchoolEvents = LinkedHashMap<DateTime, List<NTUTCalendarJson>>(equals: isSameDay);
+
+  final selectedEvents = <NTUTCalendarJson>[].obs;
+  final calendarFormat = CalendarFormat.month.obs;
+  final focusDay = DateTime.now().toLocal().obs;
+  final selectedDay = DateTime.now().toLocal().obs;
+
+  final firstDay = DateTime(1990);
+  final lastDay = DateTime(2099);
+
+  final _storedMonthSet = <DateTime>{};
+
+  String get currentCalendarLocaleString => (LanguageUtil.getLangIndex() == LangEnum.zh) ? "zh_CN" : "en_US";
+
+  List<NTUTCalendarJson> getEventsFromDay(DateTime day) => knownSchoolEvents[day] ?? const [];
+
+  bool isHoliday(DateTime day) => knownHolidays.containsKey(day);
+
+  bool isSelectingSelectedDay(DateTime day) => isSameDay(selectedDay.value, day);
+
+  void onDaySelected(DateTime selectedDay, DateTime focusedDay) {
+    if (!isSameDay(this.selectedDay.value, selectedDay)) {
+      this.selectedDay.value = selectedDay;
+      focusDay.value = focusedDay;
+      selectedEvents.value = getEventsFromDay(selectedDay);
```

Comment:
According to the comment in line 25~28, there is another small benefit of that renaming is that we don't need to add `this` to distinguish parameters and class variables.

In addition, I also suggest to rename `focusDay` to `focusedDayObs`.

Reply:
addressed in [3044425](https://github.com/NEO-TAT/tat_flutter/pull/154/commits/3044425e131a0a4948dcd3776e44b9b5aa0c06e6).